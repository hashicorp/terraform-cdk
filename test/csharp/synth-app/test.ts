// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver, onlyHcl, onlyJson } from "../../test-helper";

describe("csharp full integration test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupCsharpProject();
  });

  test("debug command", async () => {
    const { stdout } = await driver.exec(`cdktf debug --json`);
    const { cdktf, jsii, constructs } = JSON.parse(stdout);
    expect(cdktf).toBeDefined();
    expect(jsii).toBeDefined();
    expect(constructs).toBeDefined();
  });

  onlyJson("synth generates JSON", async () => {
    await driver.synth();
    expect(
      driver.synthesizedStack("csharp-simple").toString()
    ).toMatchSnapshot();
  });

  onlyHcl("synth generates HCL", async () => {
    await driver.synth();
    expect(
      driver.synthesizedStackContentsRaw("csharp-simple").toString()
    ).toMatchSnapshot();
  });
});
