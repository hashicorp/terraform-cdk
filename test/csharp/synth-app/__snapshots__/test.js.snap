// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csharp full integration test synth synth generates JSON 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"csharp-simple\\",
      \\"backend\\": \\"local\\"
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"null\\": {
        \\"version\\": \\"~> 3.1.0\\",
        \\"source\\": \\"hashicorp/null\\"
      }
    }
  },
  \\"provider\\": {
    \\"null\\": [
      {}
    ]
  },
  \\"resource\\": {
    \\"null_resource\\": {
      \\"null\\": {
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"csharp-simple/null\\",
            \\"uniqueId\\": \\"null\\"
          }
        }
      }
    }
  }
}"
`;
